modules/ballerina-http/src/main/ballerina/ballerina/net/http/request.bal

```diff
@@ -1,91 +1,235 @@
-        var valueArray =  getHeaderValueArray(headerValues, headerName);
+        var valueArray = getHeaderValueArray(headerValues, headerName);
         valueArray[lengthof valueArray] = {value:headerValue};
     }
 }
 
-@Description { value:"Gets transport headers from the request"}
-@Param { value:"req: A request message" }
-@Param { value:"headerName: The header name" }
-@Return { value:"The header values struct array for a given header name" }
-public function <Request req> getHeaders (string headerName) (HeaderValue[]) {
-    if (req.headers == null) {
+@Description {value:"Gets transport headers from the request"}
+@Param {value:"req: A request message"}
+@Param {value:"headerName: The header name"}
+@Return {value:"The header values struct array for a given header name"}
+public function <Request req> getHeaders (string headerName) (mime:HeaderValue[]) {
+    mime:Entity entity = req.getEntity(false);
+    if (entity.headers == null) {
         return null;
     }
-    var headerValues = req.headers[headerName];
+    var headerValues = entity.headers[headerName];
     if (headerValues == null) {
         return null;
     }
     return getHeaderValueArray(headerValues, headerName);
 }
 
-@Description { value:"Sets the value of a transport header"}
-@Param { value:"req: A request message" }
-@Param { value:"headerName: The header name" }
-@Param { value:"headerValue: The header value" }
+@Description {value:"Sets the value of a transport header"}
+@Param {value:"req: A request message"}
+@Param {value:"headerName: The header name"}
+@Param {value:"headerValue: The header value"}
 public function <Request req> setHeader (string headerName, string headerValue) {
-    if (req.headers == null) {
-        req.headers = {};
+    mime:Entity entity = req.getEntity(false);
+    if (entity.headers == null) {
+        entity.headers = {};
     }
-    HeaderValue[] header = [{value:headerValue}];
-    req.headers[headerName] = header;
+    mime:HeaderValue[] header = [{value:headerValue}];
+    entity.headers[headerName] = header;
 }
 
-@Description { value:"Removes a transport header from the request"}
-@Param { value:"req: A request message" }
-@Param { value:"key: The header name" }
+@Description {value:"Removes a transport header from the request"}
+@Param {value:"req: A request message"}
+@Param {value:"key: The header name"}
 public function <Request req> removeHeader (string key) {
-    if (req.headers == null) {
+    mime:Entity entity = req.getEntity(false);
+    if (entity.headers == null) {
         return;
     }
-    req.headers.remove(key);
+    entity.headers.remove(key);
 }
 
-@Description { value:"Removes all transport headers from the message"}
-@Param { value:"req: A request message" }
+@Description {value:"Removes all transport headers from the message"}
+@Param {value:"req: A request message"}
 public function <Request req> removeAllHeaders () {
-    req.headers = {};
+    mime:Entity entity = req.getEntity(false);
```

Comment:
Shall we have two different functions instead of parsing a flag. Because parsing says that this function does two things which breaks principle of a given function should only do one and only on thing.